plugins/inputs/icinga2/icinga2.go

```diff
@@ -31,36 +36,74 @@ type Icinga2 struct {
 	client *http.Client
 }
 
-type Result struct {
-	Results []Object `json:"results"`
+type ResultObject struct {
+	Results []struct {
+		Attrs struct {
+			CheckCommand string  `json:"check_command"`
+			DisplayName  string  `json:"display_name"`
+			Name         string  `json:"name"`
+			State        float64 `json:"state"`
+			HostName     string  `json:"host_name"`
+		} `json:"attrs"`
+		Name  string   `json:"name"`
+		Joins struct{} `json:"joins"`
+		Meta  struct{} `json:"meta"`
+		Type  string   `json:"type"`
+	} `json:"results"`
 }
 
-type Object struct {
-	Attrs Attribute  `json:"attrs"`
-	Name  string     `json:"name"`
-	Joins struct{}   `json:"joins"`
-	Meta  struct{}   `json:"meta"`
-	Type  ObjectType `json:"type"`
+type ResultCIB struct {
+	Results []struct {
+		Status map[string]interface{} `json:"status"`
+	} `json:"results"`
 }
 
-type Attribute struct {
-	CheckCommand string  `json:"check_command"`
-	DisplayName  string  `json:"display_name"`
-	Name         string  `json:"name"`
-	State        float64 `json:"state"`
-	HostName     string  `json:"host_name"`
+type ResultPerfdata struct {
+	Results []struct {
+		Perfdata []struct {
+			Label string  `json:"label"`
+			Value float64 `json:"value"`
+		} `json:"perfdata"`
+	} `json:"results"`
 }
 
 var levels = []string{"ok", "warning", "critical", "unknown"}
 
-type ObjectType string
-
 func (*Icinga2) SampleConfig() string {
 	return sampleConfig
 }
 
-func (i *Icinga2) GatherStatus(acc telegraf.Accumulator, checks []Object) {
-	for _, check := range checks {
+func (i *Icinga2) Init() error {
+	var err error
+	availableStatusEndpoints := []string{"ApiListener", "CIB", "IdoMysqlConnection", "IdoPgsqlConnection"}
+	err = choice.CheckSlice(i.Status, availableStatusEndpoints)
+	if err != nil {
+		return err
+	}
```

Comment:
Some more context will be easier to understand the error
```suggestion
	err = choice.CheckSlice(i.Status, availableStatusEndpoints)
	if err != nil {
		return fmt.Errorf("config option 'status': %w", err)
	}
```

Reply:
Fixed.